### DISCUSSION TITLE
Do I have to use set_default_keymaps()?

### DISCUSSION THREAD

**Discussion by gtnbssn** [ggandor/leap.nvim#56] [Q&A] [answered]
After installation and setup, `s` wasn't remapped to use leap and kept doing its normal Vim thing.

I found that doing `set_default_keymaps()` before calling the setup function helped. Is it supposed to be this way?

Or did i miss something maybe?

This is how i am doing it now: https://github.com/gtnbssn/dotfiles/blob/main/nvim/.config/nvim/lua/pluginsConf/leap.lua

It stops working if i remove the call to set_default_keymaps().

**Comment by ggandor** [comment, marked as answer]
Yes. As it is described in the docs, `setup` just for updating multiple values in the configuration table (which you could also do manually). `set_default_keymaps` is unrelated, you should call it separately.

**Reply by gtnbssn** [reply]
Thanks! I didn't realize this was necessary. The doc says "No extra steps needed, besides *optionally* setting the default keymaps" (emphasis mine) so i didn't realize it wouldn't remap `s` without this.

I would suggest making it more explicit, but it seems most users have figured it out!

**Reply by ggandor** [reply]
Optionally, since you might not want the defaults, and want to define custom mappings. What would be the point of `set_default_keymaps` at all, if the plugin would do that automatically anyway? But I'll try to rephrase that, to be extra clear :)

**Reply by gtnbssn** [reply]
I think what is confusing me is that the documentation doesn't show how to map the trigger keys in the setup function. But when i read it, it sounds like `s` should automatically have been remapped for leap.

Right now it looks to me like i *have* to call set_default_keymaps(), then define custom mappings. Otherwise the plugin doesn't work. But when i define custom mappings, there is nothing for the trigger keys.

Sorry if i am missing something!

**Reply by chaitanyabsprip** [reply]
I had the same impression.

**Reply by gtnbssn** [reply]
It looks like things changed 2 months ago. The documentation is clearer now, one needs to call `add_default_mappings()`

Not sure how this differs from `set_default_keymaps()`, which still exists?

**Reply by ggandor** [reply]
The latter is deprecated (and it adds _different_ mappings than the new one).
